nixos/modules/services/web-apps/nextcloud.nix

```diff
@@ -268,7 +268,14 @@ in {
     };
 
 
-    config = {
+    config = let
+      uiClashWarning = ''
+        <warning>
+          <para>Setting this option will prevent changing this via the
+          Nextcloud UI, but it will still appear editable in the UI.</para>
```

Comment:
Actually it doesn't: when messing around in the mail settings, stuff like `mail_domain`/`mail_from_address`/`mail_sendmailmode` appear with the values from the UI in `config.php`. Yes, things from `override.config.php` take precedence, but since all of these settings are optional, this can still cause unwanted side-effects if parts of the config come from the UI and another part from the Nix expression.

Suggestions what to do:

* Force all `mail`-options to be declared if enabled
* Perhaps even write a patch to disable the controls in the UI.

To be frank, I'm not sure anymore if that's a reasonable thing to do. cc @aanderse @nlewo @ajs124 @beardhatcode for opinions.

Reply:
Some software really wants to be configured via the ui. We do something similarly hacky with the [nzbget](https://github.com/NixOS/nixpkgs/blob/nixos-22.05/nixos/modules/services/misc/nzbget.nix) module... but the module is tiny, the code is easy to understand, and overall it is a pretty simple piece of software compared to `nextcloud`. Not directly comparable, but maybe my comments provide at least a little bit of value.

:man_shrugging: